src/app/actions.js

```diff
@@ -38,3 +39,15 @@ export function storeVisitURL(visitURL) {
     payload: visitURL
   };
 }
+
+/**
+  Flush localStorage login keys
+*/
+export function LogOut() {
```

Comment:
In a basic `action creator` you don't really want any side effects and instead only try and generate the `message object`. To get around this limitation we use the `thunks` redux plugin which allows us to perform arbitrary side effects. It's something you want to try and limit as it can make the code really hard to reason about if abused. But in this case it's what you want.

A more idiomatic approach to this would be something like we've done in https://github.com/canonical-web-and-design/jaas-dashboard/blob/master/src/juju/actions.js#L56

This code would become something like...

```JavaScript
export function clearUserData() {
  return {
    type: actionsList.clearUserData
  }
}

export function logOut() {
  return async function thunk(dispatch, getState) {
    localStorage.removeItem("identity");
    localStorage.removeItem("https://api.jujucharms.com/identity");;
    dispatch(clearUserData());
  };
}
```